### DISCUSSION TITLE
Bugfix: please fix broken timer for mark of darkness

### DISCUSSION THREAD

**Discussion by robertino129** [runelite/runelite#18123] [Q&A]
Broken timer: mark of darkness
I think the file that needs to be changed is: https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

I think it needs at least this change
line 867
		
```
	                else if (message.endsWith(MARK_OF_DARKNESS_MESSAGE))
			{
				Item weapon = how to inject container to get weapon;
				if (weapon != null && ItemID.PURGING_STAFF.contains(weapon.getId())) {
					createGameTimer(MARK_OF_DARKNESS, 5 * Duration.of(magicLevel, RSTimeUnit.GAME_TICKS));
				} else {
					createGameTimer(MARK_OF_DARKNESS, Duration.of(magicLevel, RSTimeUnit.GAME_TICKS));
				}
			}

```

If anyone with more knowledge about the codebase could figure out how to inject and get the equipped weapon and fix it, that would be awesome.

Thanks.

**Comment by YvesW** [comment]
Do you mean that the increased mark of darkness timer when wearing the purging staff has not yet been implemented?

**Reply by robertino129** [reply]
Not in the default timers shipped together with runelite.

**Reply by YvesW** [reply]
Should be fixed by https://github.com/runelite/runelite/pull/18343
Keep in mind that it still needs to be reviewed by a maintainer, so it might take a while to show up in the client. Just wait patiently.
